### ISSUE TITLE
[BUG] i915 gpu not detected for opencl

### ISSUE THREAD

**Issue by izenn** [linuxserver/docker-mods#871]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Name of mod

Opencl-Intel

### Name of base container

jellyfin

### Current Behavior

when on kernel 6.8 there is a bug in the intel-compute runtime that prevents opencl from getting the gpu:

```
$ docker exec -it jellyfin bash
root@jellyfin:/# clinfo
Number of platforms                               0
```
in the jellyfish logs the following is given as output:
```
[AVHWDeviceContext @ 0x586d0f248480] Failed to get number of OpenCL platforms: -1001.
Device creation failed: -19.
Failed to set value 'opencl=ocl@va' for option 'init_hw_device': No such device
Error parsing global options: No such device
```

### Expected Behavior

to fix this, the following needs to be put in the docker-compose.yml:
```
    environment:
      - NEOReadDebugKeys=1
      - OverrideGpuAddressSpace=48
```
(upstream bug info can be found here: https://github.com/intel/compute-runtime/issues/710)

### Steps To Reproduce

1) create docker-compose.yml using examples found https://docs.linuxserver.io/images/docker-jellyfin/
2) enable hardware transcoding with tone mapping enabled

### Environment

```markdown
- OS: docker on top of linux system with a 6.8 kernel
```

### CPU architecture

x86-64

### Docker creation

```bash
---
services:
  jellyfin:
    image: lscr.io/linuxserver/jellyfin:latest
    container_name: jellyfin
    hostname: jellyfin
    environment:
      - PUID=1000
      - PGID=1000
      - DOCKER_MODS=linuxserver/mods:jellyfin-opencl-intel
      - NEOReadDebugKeys=1
      - OverrideGpuAddressSpace=48
    volumes:
      - /data/jellyfin/config:/config
      - /dev/shm:/config/data/transcodes
      - /data/jellyfin/cache:/config/cache
      - /data/nfs/Media/TV:/data/TV
      - /data/nfs/Movies:/data/Movies
    devices:
      - /dev/dri:/dev/dri
    networks:
      - jellyfin
    ports:
      - 8096:8096
      - 8920:8920 #optional
      - 7359:7359/udp #optional
      - 1900:1900/udp #optional
    restart: always

networks:
  jellyfin:
    driver: bridge
    name: jellyfin
    external: true
```

### Container logs

```bash
see above for logs and work around
```

**Comment by eero-t** [comment]
> when on kernel 6.8 there is a bug in the intel-compute runtime that prevents opencl from getting the gpu

Or newer kernel (or drm integration drm-tip 6.7 kernel).

>      - NEOReadDebugKeys=1
>      - OverrideGpuAddressSpace=48

This may impact compute SVM capability.  Better to upgrade to a fixed compute-runtime version, if possible.

**Comment by izenn** [comment]
a fixed compute-runtime is the preferred solution, however it is not available at this time (see the known issues section of the latest release: https://github.com/intel/compute-runtime/releases/tag/24.09.28717.12)

**Comment by eero-t** [comment]
That's 3 weeks old.  I was thinking of the more recent tag including the fix:
https://github.com/intel/compute-runtime/compare/24.09.28717.12...24.09.28717.14

**Comment by izenn** [comment]
this is very welcome news.  hopefully they put out a release soon for those who don't want to or don't feel comfortable compiling.

**Comment by mLgz0rn** [comment]
@izenn are you able to play content when using opencl for tonemapping with the workaround?
My player just seem to get stuck trying to play and fails in the end.

**Comment by izenn** [comment]
I have not had issues at all (other than an occasional hiccup because i'm transferring everything over the network rather than locally and my nas is ancient).  

Here is my docker-compose and my hardware acceleration settings:
```
---
services:
  jellyfin:
    image: lscr.io/linuxserver/jellyfin:latest
    container_name: jellyfin
    hostname: jellyfin
    environment:
      - PUID=1000
      - PGID=1000
      - UMASK=002
      - DOCKER_MODS=linuxserver/mods:jellyfin-opencl-intel
      - JELLYFIN_PublishedServerUrl=http://192.168.1.101:8096
      - NEOReadDebugKeys=1
      - OverrideGpuAddressSpace=48
    volumes:
      - /data/jellyfin/config:/config
      - /dev/shm:/config/data/transcodes
      - /data/jellyfin/cache:/config/cache
      - /data/nfs/Media/Misc:/data/Misc
      - /data/nfs/Media/TV:/data/TV
      - /data/nfs/Movies:/data/Movies
    devices:
      - /dev/dri:/dev/dri
    networks:
      - jellyfin
    ports:
      - 8096:8096
      - 8920:8920 #optional
      - 7359:7359/udp #optional
      - 1900:1900/udp #optional
    restart: always

networks:
  jellyfin:
    driver: bridge
    name: jellyfin
    external: true
```
<img width="876" alt="Screenshot 2024-04-16 at 8 29 38 AM" src="https://github.com/linuxserver/docker-mods/assets/744288/71b412ac-f381-47e4-b4cc-0f4a7ca8ee15">
<img width="854" alt="Screenshot 2024-04-16 at 8 29 51 AM" src="https://github.com/linuxserver/docker-mods/assets/744288/2ea388ca-0aa3-4544-a214-423505b2f6f8">
<img width="847" alt="Screenshot 2024-04-16 at 8 30 00 AM" src="https://github.com/linuxserver/docker-mods/assets/744288/d6afbd13-b230-4357-8767-3e90a334675a">
<img width="835" alt="Screenshot 2024-04-16 at 8 30 10 AM" src="https://github.com/linuxserver/docker-mods/assets/744288/e419c06f-dfe8-4488-8432-f2342f3bf2c1">
<img width="844" alt="Screenshot 2024-04-16 at 8 30 18 AM" src="https://github.com/linuxserver/docker-mods/assets/744288/1c1d1911-9d7d-46c7-bcba-8dbdd8f8bf68">

**Comment by mLgz0rn** [comment]
Hmm weird, I downgraded my kernel again to 6.7.5 because I couldnt get it to work.
Are you using jellyfin's ffmpeg 5.x or 6.x?  Don't know if that could have anything to say.
Or if its and issue with onboard vs. dedicated gpu

I'm on an Intel Arc A380 using jellyfin-ffmpeg 6.x

**Comment by izenn** [comment]
i'm using what was provided by the docker image/mod.  i typically try to leave things like that as default as possible

**Comment by mLgz0rn** [comment]
New release is out which should fix the issue https://github.com/intel/compute-runtime/releases/tag/24.13.29138.7

**Comment by mLgz0rn** [comment]
Just fyi, I have figured out that my current issue is not an opencl issue anymore but an i915 driver bug issue that appeared in the LTS kernel 6.6.26+ and the stable kernel 6.8.5+, just my luck it happened almost at the same time then :'D

**_The LTS kernel 6.6.26+ and the stable kernel 6.8.5+ have unresolved i915 driver bugs, which break HDR/DV tone-mapping on Intel Gen 12.5 DG2 / ARC A-series GPUs. If you are affected, please refrain from upgrading to those kernel versions._**

https://github.com/jellyfin/jellyfin/issues/11380

**Comment by mLgz0rn** [comment]
Kernel 6.8.10 and 6.9+ has the fixes for the i915 driver bugs included.
So everything seems to be working now.
For my sake this issue can be closed :D

**Comment by izenn** [comment]
looks to be fixed in intel-compute 24.13.29138.7 (https://github.com/intel/compute-runtime/releases/tag/24.17.29377.6)
